### PR TITLE
feat: [LI-50] implement desktop ui for candidate select interview time

### PR DESCRIPTION
**Pull request by lulu0729**
### Issue ticket number and link
[LI-50](https://predictivehire.atlassian.net/browse/LI-50)
### Figma Screenshot
image

### Describe main changes

https://github.com/PredictiveHire/phapp-schedule-link/assets/34208317/722bc27d-9528-4b9f-912b-ed26d061557b
- implement desktop ui for candidate select interview time 

### Checklist in Local 
- [x] I have extract component business logic into custom hook
- [x] I have added unit tests
- [ ] I have integrated with backend api and done a e2e test locally
- [ ] I have updated the graphql query types

### Checklist in Dev
- [ ] I have deployed to dev
- [ ] I have integrated with backend api and done a e2e test

### Checklist before requesting a review
- [ ] I have done the **code format** using `pnpm lint:fix` and `pnpm format:fix`
- [ ] I have **rebase squash** current branch
- [ ] I have confirmed **Naming Convention** with team if needed.
- [ ] I have done a **self review** in GitHub.

### Checklist in Qa Environment
- [ ] I have deployed to qa and let QA team do a test
- [ ] I have fix the QA bugs after QA team finish testing
- [ ] I have put the screenshot for Qa environment integration test below:

### Checklist in Sandbox Environment
- [ ] I have deployed to sandbox and let QA team do test
- [ ] I have fix the Sandbox Bugs after QA team finish testing
- [ ] I have put the screenshot for Qa environment integration test below:

### Checklist after Merge
- [ ] I have known the dependencies of release deployment of current PR.
- [ ] I have reviewed the release pipe from team lead to secure the production deployment.
- [ ] I have update the Jira tickets status after all task has been finished.

[LI-50]: https://predictivehire.atlassian.net/browse/LI-50?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+                  {slot}
+                </Radio.Button>
+              ))}
+            </Radio.Group>
+          </div>
+        </section>
+      </section>
+      <Button
+        className={cn([
+          "buttonBook mt-4 w-full text-base font-normal",
```
Comment (martin-luo) [inline src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx:97]: Same question here, why we need `buttonBook` here?
```suggestion
          "mt-4 w-full text-base font-normal",
```

I saw you use this in the css module to locate the button and then change the styles, but I don't think that's how it should work
image

If you do need another class to locate the button or change the precedence, you can try to take it out of `:global()`, and then use it as `styles.buttonBook` in the component
image
image

By doing so, we can make sure each of these selectors are well-scoped

File: src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/DesktopCandidateSelectInterviewTime.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  jobRequisitionName: "job requisition name",
+  interviewMode: LIInterviewMode.IN_PERSON,
+  interviewAddress: "216 Lower Heidelberg Rd, Ivanhoe East VIC 3079",
+}
+
+export const DesktopCandidateSelectInterviewTime = () => {
+  return (
+    <div
+      data-testid="desktop-candidate-select-interview-time"
+      className="h-full overflow-y-auto bg-gray-100 px-11 py-11"
```
Comment (bennyli519) [inline src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/DesktopCandidateSelectInterviewTime.tsx:20]:
```suggestion
      className="h-full overflow-y-auto bg-gray-100 p-11"
```

File: src/pages/schedule-interview/components/InterviewInfo/__test__/InterviewInfo.test.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      interviewMode: LIInterviewMode.IN_PERSON,
+      interviewAddress: "123 Main St",
+      interviewLink: "https://example.com",
+    },
+  }
+
+  it("should renders correctly with in-person interview mode", () => {
+    render(<InterviewInfo interviewInfo={{ ...baseProps.interviewInfo, interviewMode: LIInterviewMode.IN_PERSON }} />)
+
+    expect(screen.getByAltText("Logo")).toHaveAttribute("src", "test-logo.png")
```
Comment (martin-luo) [inline src/pages/schedule-interview/components/InterviewInfo/__test__/InterviewInfo.test.tsx:22]: I think we prefer `screen.getByRole("img")` over `screen.getByAltText`

File: src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+  } = useSelectTimeSlot()
+  const { formatTimeSlots } = useTimeSlots()
+  const currentDate = formatDayjs(interviewDate)
+  const timeSlots = formatTimeSlots(interviewDates, currentDate)
+  const userTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone
+
+  const headerRender = ({ value, onChange }: { value: Dayjs; onChange: (date: Dayjs) => void }) => (
+    <div className="mb-4 flex items-center justify-between px-5">
+      <LeftOutlined
+        onClick={() => onChange(value.clone().subtract(1, "month"))}
```
Comment (martin-luo) [inline src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx:35]: define a function for this, to improve the maintainability and readability

File: src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+        className="text-secondary cursor-pointer text-xl"
+      />
+      <span className="px-2 text-base font-semibold text-black">{`${value.format("MMMM")} ${value.year()}`}</span>
+      <RightOutlined
+        onClick={() => onChange(value.clone().add(1, "month"))}
```
Comment (martin-luo) [inline src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx:40]: define a function for this, to improve the maintainability and readability

File: src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    </div>
+  )
+
+  const fullCellRender = (value: Dayjs) => (
+    <div
+      className={cn([
+        "ant-picker-cell-inner ant-picker-calendar-date rounded-full text-center",
+        isInterviewDate(value) ? "text-secondary bg-[rgba(0,0,0,0.04)] text-base" : "text-sm text-black/50",
+        formatDayjs(initialInterviewDate) === formatDayjs(value) &&
+          currentDate === formatDayjs(value) &&
```
Comment (martin-luo) [inline src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx:52]: define a variable for this to improve the maintainability and readability

File: src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+          "bg-sapia-pink",
+      ])}
+      onClick={() => handleDateChange(value)}
+    >
+      <div className="rounded-full">{value.date()}</div>
+    </div>
+  )
+
+  return (
+    <div data-testid="desktop-interview-time-slot-booking" className={cn([styles.interviewTimeSlotBookingWrapper])}>
```
Comment (martin-luo) [inline src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/components/DesktopInterviewTimeSlotBooking/DesktopInterviewTimeSlotBooking.tsx:62]:
```suggestion
    <div data-testid="desktop-interview-time-slot-booking" className={styles.interviewTimeSlotBookingWrapper}>
```

File: src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/__test__/DesktopCandidateSelectInterviewTime.test.tsx [hunk 8]
```diff
@@ -0,0 +1,33 @@
+import { render, screen } from "@testing-library/react"
+import React from "react"
+
+import { DesktopCandidateSelectInterviewTime } from "../DesktopCandidateSelectInterviewTime"
+
+jest.mock("../../../../assets/images/ww-logo.svg", () => "logoSvgMock")
```
Comment (martin-luo) [inline src/pages/schedule-interview/components/DesktopCandidateSelectInterviewTime/__test__/DesktopCandidateSelectInterviewTime.test.tsx:6]: why do we mock an image here?

File: src/pages/schedule-interview/hooks/useCalendarHeader.ts
Comment (martin-luo) [inline src/pages/schedule-interview/hooks/useCalendarHeader.ts:1]: Please add unit test for this hook

**Review by martin-luo — APPROVED** [review]
Looks good to me, just a minor comment

**Review by bennyli519 — APPROVED** [review]

**Pull request merged**